adapters/interlocked_win32.c

```diff
@@ -1,4 +1,318 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-#include "azure_c_util/interlocked.h"
\ No newline at end of file
+#include "azure_c_util/interlocked.h"
+#include "windows.h"
```

Comment:
This include should go above. We normally include all standard headers, then system headers like windows.h, then project headers.

Also, you should include stdint.h for the intX_t types.